awscli/completer.py

```diff
@@ -19,19 +19,49 @@
 
 class Completer(object):
 
-    def __init__(self):
-        self.driver = awscli.clidriver.create_clidriver()
-        self.main_hc = self.driver.create_help_command()
-        self.main_options = self._documented(self.main_hc.arg_table)
-        self.cmdline = None
-        self.point = None
-        self.command_hc = None
-        self.subcommand_hc = None
-        self.command_name = None
-        self.subcommand_name = None
-        self.current_word = None
-        self.previous_word = None
-        self.non_options = None
+    def __init__(self, driver=awscli.clidriver.create_clidriver()):
```

Comment:
You generally don't want to have default mutable arguments.  The common pattern is to default this to `None`, then check if `driver is None` and create a default value in that scenario.  Otherwise multiple instances may unknowingly share the same argument value and mutations in one instance will be reflected in instances of the other `Completer`.